src/Kuzzle.ts

```diff
@@ -259,14 +314,18 @@ class Kuzzle extends KuzzleEventEmitter {
 
     if (protectedEvent) {
       if ( protectedEvent.lastEmitted
-        && protectedEvent.lastEmitted > now - this.eventTimeout)
-      {
+        && protectedEvent.lastEmitted > now - this.eventTimeout
+      ) {
         return false;
       }
       protectedEvent.lastEmitted = now;
     }
 
-    super.emit(eventName, ...payload);
+    return this._superEmit(eventName, ...payload);
+  }
+
+  _superEmit (eventName, ...payload) {
```

Comment:
What's the purpose of this function? :thinking: 

(same goes for  [_superAddListener()](https://github.com/kuzzleio/sdk-javascript/pull/526/files#diff-5a8206688414a52fe7efb0e1bef73b1cR421))

Reply:
To simplify the test.. It's was much more easier to just mock `_superEmit` than to mock the import of `KuzzleEventEmitter` (it didn't work after the TS conversion)

Reply:
You normally don't need to do that, you can directly mock the parent class (e.g. https://github.com/kuzzleio/kuzzle/blob/master/test/core/security/userRepository.test.js#L90 )

I won't block the PR for that, though.

Reply:
I got a `TypeError: Attempted to wrap emit which is already wrapped` when I try to wrap it that way but when I remove the line I got `expected 'emit' ...  to be sinon spy or spy call`